python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/test_catboost.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) 2023-2025, NVIDIA CORPORATION.
+import numpy as np
+import pandas as pd
+import pytest
+from catboost import CatBoostClassifier, CatBoostRegressor, Pool
+from sklearn.datasets import make_classification, make_regression
+
+rng = np.random.default_rng(seed=42)
+
+
+def assert_catboost_equal(expect, got, rtol=1e-7, atol=0.0):
+    if isinstance(expect, (tuple, list)):
+        assert len(expect) == len(got)
+        for e, g in zip(expect, got):
+            assert_catboost_equal(e, g, rtol, atol)
+    elif isinstance(expect, np.ndarray):
+        np.testing.assert_allclose(expect, got, rtol=rtol, atol=atol)
+    elif isinstance(expect, pd.DataFrame):
+        pd.testing.assert_frame_equal(expect, got)
+    elif isinstance(expect, pd.Series):
+        pd.testing.assert_series_equal(expect, got)
+    else:
+        assert expect == got
+
+
+pytestmark = pytest.mark.assert_eq(fn=assert_catboost_equal)
+
+
+@pytest.fixture
+def regression_data():
+    X, y = make_regression(n_samples=100, n_features=10, random_state=42)
+    return pd.DataFrame(X), pd.Series(y)
+
+
+@pytest.fixture
+def classification_data():
+    X, y = make_classification(
+        n_samples=100, n_features=10, n_classes=2, random_state=42
+    )
+    return pd.DataFrame(X), pd.Series(y)
+
+
+def test_catboost_regressor_with_dataframe(regression_data):
+    X, y = regression_data
+    model = CatBoostRegressor(iterations=10, verbose=0)
+    model.fit(X, y)
+    predictions = model.predict(X)
+    return predictions
+
+
+def test_catboost_regressor_with_numpy(regression_data):
+    X, y = regression_data
+    model = CatBoostRegressor(iterations=10, verbose=0)
+    model.fit(X.values, y.values)
+    predictions = model.predict(X.values)
+    return predictions
```

Comment:
Sorry in advance, I'm not that familiar with these tests but... I'm surprised to see `pytest` test cases with a `return` statement. What is the interaction between these test cases and this a few lines up?

```python
pytestmark = pytest.mark.assert_eq(fn=assert_catboost_equal)
```

Did you mean for there to be some kind of testing assertion here? Or does that custom marker somehow end up invoking that function and comparing the output of the test case with `pandas` inputs to its output with `cudf` inputs?

Reply:
The assertion function is used to check that results from "cudf.pandas on" and "cudf.pandas off" are equal. The logic to handle that is in the conftest file.

Reply:
> the conftest file

Thanks, I wasn't sure which `conftest.py` to look in. I guess this is the relevant piece, yeah?

https://github.com/rapidsai/cudf/blob/300d8165dc5d27a197aa655e34781713465c41ec/python/cudf/cudf_pandas_tests/third_party_integration_tests/tests/conftest.py#L130-L148

Reply:
Yup!